samples/place-autocomplete-data-session/index.ts

```diff
@@ -5,97 +5,146 @@
+    google.maps.event.addListener(innerMap, 'bounds_changed', async () => {
+        request.locationRestriction = innerMap.getBounds();
+        request.origin = innerMap.getCenter();
+    });
+
     inputElement.addEventListener('input', makeAutocompleteRequest);
-    refreshToken(request);
 }
 
 async function makeAutocompleteRequest(inputEvent) {
+    // To avoid race conditions, store the request ID and compare after the request.
+    const requestId = ++newestRequestId
+
+    const { AutocompleteSuggestion } = (await google.maps.importLibrary(
+        'places'
+    )) as google.maps.PlacesLibrary;
+
     // Reset elements and exit if an empty string is received.
-    if (inputEvent.target.value == '') {
-        titleElement.innerText = '';
+    if (!inputEvent.target ||
+        inputEvent.target.value === '') {
+        titleElement.textContent = '';
         resultsContainerElement.replaceChildren();
         return;
     }
 
     // Add the latest char sequence to the request.
-    request.input = inputEvent.target.value;
-
-    // To avoid race conditions, store the request ID and compare after the request.
-    const requestId = ++newestRequestId;
+    request.input = (inputEvent.target as HTMLInputElement).value;
 
     // Fetch autocomplete suggestions and show them in a list.
-    // @ts-ignore
-    const { suggestions } = await google.maps.places.AutocompleteSuggestion.fetchAutocompleteSuggestions(request);
+    const { suggestions } =
+        await AutocompleteSuggestion.fetchAutocompleteSuggestions(request);
 
     // If the request has been superseded by a newer request, do not render the output.
     if (requestId !== newestRequestId) return;
 
-    titleElement.innerText = `Query predictions for "${request.input}"`;
+    titleElement.innerText = `Place predictions for "${request.input}"`;
 
     // Clear the list first.
     resultsContainerElement.replaceChildren();
 
     for (const suggestion of suggestions) {
         const placePrediction = suggestion.placePrediction;
 
+        if (!placePrediction) {
+            continue
+        }
+
         // Create a link for the place, add an event handler to fetch the place.
-        const a = document.createElement('a');
-        a.addEventListener('click', () => {
-            onPlaceSelected(placePrediction!.toPlace());
+        // We are using a button element to take advantage of its a11y capabilities.
+        const link = document.createElement('button');
+        link.addEventListener('click', () => {
+            onPlaceSelected(placePrediction.toPlace());
         });
-        a.innerText = placePrediction!.text.toString();
+        link.textContent = placePrediction.text.toString();
+        link.classList.add('link-button');
 
         // Create a new list item element.
         const li = document.createElement('li');
-        li.appendChild(a);
+        li.appendChild(link);
         resultsContainerElement.appendChild(li);
```

Comment:
I would just call it a button (or placeButton) for simplicity


```suggestion
        // Create a button for the place, add an event handler to fetch the place.
        const placeButton = document.createElement('button');
        placeButton.addEventListener('click', () => {
            onPlaceSelected(placePrediction.toPlace());
});
        placeButton.textContent = placePrediction.text.toString();
        placeButton.classList.add('place-button');

        // Create a new list item element.
        const li = document.createElement('li');
        li.appendChild(link);
        resultsContainerElement.appendChild(li);
```